app/components/delivery-preview.js

```diff
@@ -0,0 +1,27 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  delivery: null,
+  emailMessage: null,
+  errors: null, // Default value
+  onFail: (/*errors*/) => {}, // Defaults to an empty operation, but should be provided as an (action 'previewFailed')
+  didInsertElement() {
+    this._super(...arguments);
+    this.generatePreview();
+  },
+
+  generatePreview() {
+    this.set('emailMessage', null); // causes loading indicator to appear
+    // Delivery may be a model object that's not yet loaded (e.g. a relationship)
+    let delivery = this.get('delivery');
+    if(delivery.get('isLoaded')) {
+      delivery = Ember.RSVP.resolve(delivery);
+    }
+    delivery.then((loadedDelivery) => {
```

Comment:
Had to read this a couple times to figure out what is going on: 
If delivery is not a promise make it one.

Reply:
I updated the comment around this. I still don't like how this works but have been digging in to understand why it's not working.

The underlying issue is that when I create a`{{delivery-preview}}` component in `resend-confirm.hbs`, the `delivery` object I provide is `model.delivery`, where `model` is a `duke-ds-project-transfer`.

When `delivery-preivew.js` accesses that `delivery`, it's a `DS.PromiseObject`. `DS.PromiseObject`s do not a have a `preview()` method, so the promise needs to be resolved to a "real" delivery before calling `preview()`. If I call `delivery.get('preview')()`, it does seem find the function, but then it's not bound to the right `this`.

Using `{{delivery-preview}}` from `deliveries/new/confirm.hbs` doesn't have this problem. That `delivery` object is the model for the route, so by the time it gets to the template, it's fully loaded and resolved.

Perhaps the best (and conventional) solution is for the `deliveries/show/confirm-resend` route to return the `delivery` as its model. The transfer can be accessed as a model from `deliveries/show`